src/core/context/index.js

```diff
@@ -87,7 +87,7 @@ export function rediscoverChildren(node) {
  *
  * @param {Node} node The target node.
  * @param {IContextProp<T, ?>} prop
- * @param {function(T):void} setter
+ * @param {function(*):*} setter
```

Comment:
note: these two setter types were just flat out wrong, and I couldn't easily figure out what they _should_ be. setting to any for now.

Reply:
You may have landed on it in a way that was internally consistent within `src/core/context`, but its consumers were not being type checked. They are utterly wrong w.r.t. usage within PBE

Reply:
You're right, I focused entirely on internal consistency. SGTM

Reply:
I just tried to understand it again and utterly gave up. This is really really hairy code